Adding documentation for output.libraryTarget 'system' (#3014)
webpack/webpack.js.org: src/content/configuration/output.md

};
```

`libraryTarget: 'system'` - This will expose your library as a [`System.register`](https://github.com/systemjs/systemjs/blob/master/docs/system-register.md)

> module. This feature was first released in [webpack@4.30.0](https://github.com/webpack/webpack/releases/tag/v4.30.0).

Reviewer: Sorry for missing this from the first time but preferred way to mention a version is `webpack 4.30.0`, you can find the relevant info here: https://webpack.js.org/contribute/writers-guide/#typesetting

```suggestion
module. This feature was first released in [webpack 4.30.0](https://github.com/webpack/webpack/releases/tag/v4.30.0).
```

};
```

> `libraryTarget: 'system'` - This will expose your library as a [System.register](https://github.com/systemjs/systemjs/blob/master/docs/system-register.md)

Reviewer: Suggest we wrap real code call references with \`
```suggestion
`libraryTarget: 'system'` - This will expose your library as a [`System.register`](https://github.com/systemjs/systemjs/blob/master/docs/system-register.md)
```

module. This feature was first released in [webpack@4.30.0](https://github.com/webpack/webpack/releases/tag/v4.30.0).

> System modules require that a global variable `System` is present in the browser when the webpack bundle is executed. Compiling to System.register format allows

Reviewer: We don't split text string with newlines unless it needs a new paragraph, which results in a new `<p>` after compilation.
```suggestion
System modules require that a global variable `System` is present in the browser when the webpack bundle is executed. Compiling to `System.register` format allows you to `System.import('/bundle.js')` without additional configuration and have your webpack bundle loaded into the System module registry.
```

> you to `System.import('/bundle.js')` without any configuration and have your webpack bundle loaded into the System module registry.

Reviewer: ```suggestion
```

> Example config:

Reviewer: This is not necessary:
```suggestion
```

```javascript
module.exports = {
  //...
  output: {
    libraryTarget: 'system'
  }
};
```

Output:

```javascript
System.register([], function(_export) {
  return {
    setters: [],
    execute: function() {
      // ...
    },
  };
});
```

> By adding `output.library` to a bundle compiled to `system` format, the output bundle will have the library name as an argument to System.register:

Reviewer: ```suggestion
By adding `output.library` to configuration in addition to having `output.libraryTarget` set to `system`, the output bundle will have the library name as an argument to `System.register`:
```
